src/content/en/2021/caching.md

```diff
@@ -1,20 +1,488 @@
+   <td>The response is intended for a specific user and should not be stored by shared caches such as CDNs.</td>
+  </tr>
+  <tr>
+   <td><code class="no-wrap">immutable</code></td>
+   <td>Indicates that the cached entry will never change during its TTL and that revalidation is not necessary.</td>
+  </tr>
+</table>
+
+The `max-age` directive is the most commonly found with 62.2% of Desktop requests including a `Cache-Control` response header with this directive.
+
+{{ figure_markup(
+  image="cache-control-directives.png",
+  caption="Usage of `Cache-Control` directives.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=1359015817&format=interactive",
+  height=436,
+  sheets_gid="1944529311",
+  sql_file="cache_control_directives.sql"
+) }}
+
+Compared to 2020, `max-age` adoption increased by 2% on Desktop, along with most of the top seven directives in the above chart.
+
+The `immutable` directive is relatively new and can significantly improve cacheability for certain types of requests. However, it is still only supported by a few browsers, and we see most requests coming from from Host Networks like *Wix* with 16.4%, *Facebook* with 8.6%, *Tawk* with 2.8%, and *Shopify* with 2.4%.
+
+The most misused `Cache-Control` directive continues to be `set-cookie`, used for 0.07% of total *directives* for Desktop and 0.08% for Mobile. However, we are pleased to see a meaningful 0.16% reduction of usage from 2020.
+
+When we take a look when `no-cache`, `max-age=0` and `no-store` are used together, we also see a growing trend year after year, in which `no-store` is specified with either/both of `no-cache` and `max-age=0`, the `no-store` directive takes precedence, and the other directives are ignored. Driving more awareness around using these *directives* during larger conferences could help avoid accidentally wasted bytes.
+
+Fun fact: The most common `max-age` value is 30 days, and the largest value is 51 trillion years.
+
+{{ figure_markup(
+  caption="TODO",
+  content="51 trillion years",
+  classes="medium-number",
+  sheets_gid="529870849",
+  sql_file="todo.sql"
+) }}
+
+### `304` Not Modified status
+
+When it comes to size, `304 Not Modified` responses are much smaller than `200 OK` responses, so it follows that page performance can be sped up by only delivering the necessary size of data. This is where correctly using conditional requests comes in because revalidation, and data savings can be done by using either an `Etag` or `Last-Modified` header.
+
+The `Last-Modified` response header works in conjunction with the `If-Modified-Since` request header to let the browser know if any changes have been made to the requested file.
+
+{{ figure_markup(
+  image="http-304-by-caching-strategy.png",
+  caption="HTTP 304 response rate by caching strategy.",
+  description="TODO",
```

Comment:
description="A bar chart of HTTP 304 response rate by caching strategy. The usage for desktop ranges from 18% for Expected with `If-Modified-Since`, 94% for Actual with `If-Modified-Since`, 84% for Expected with `ETag`, and 88% for Actual with `ETag`. Usage for mobile ranges from 13% for Expected with `If-Modified-Since`, 86% for Actual with `If-Modified-Since`, 91% for Expected with `ETag`, and 88% for Actual with `ETag` ",